cuda_core/tests/test_stream.py

```diff
@@ -6,68 +6,68 @@
-from cuda.core.experimental._event import Event, EventOptions
-from cuda.core.experimental._device import Device
+from cuda.core.experimental import Device, Stream, StreamOptions
+from cuda.core.experimental._stream import LEGACY_DEFAULT_STREAM, PER_THREAD_DEFAULT_STREAM, default_stream
+from cuda.core.experimental._event import Event
 import pytest
 
 def test_stream_init():
     with pytest.raises(NotImplementedError):
         Stream()
 
-def test_stream_init_with_options():
-    stream = Stream._init(options=StreamOptions(nonblocking=True, priority=0))
+def test_stream_init_with_options(init_cuda):
+    stream = Device().create_stream(options=StreamOptions(nonblocking=True, priority=0))
     assert stream.is_nonblocking is True
     assert stream.priority == 0
 
-def test_stream_handle():
-    stream = Stream._init(options=StreamOptions())
+def test_stream_handle(init_cuda):
+    stream = Device().create_stream(options=StreamOptions())
     assert isinstance(stream.handle, int)
 
-def test_stream_is_nonblocking():
-    stream = Stream._init(options=StreamOptions(nonblocking=True))
+def test_stream_is_nonblocking(init_cuda):
+    stream = Device().create_stream(options=StreamOptions(nonblocking=True))
     assert stream.is_nonblocking is True
 
-def test_stream_priority():
-    stream = Stream._init(options=StreamOptions(priority=0))
+def test_stream_priority(init_cuda):
+    stream = Device().create_stream(options=StreamOptions(priority=0))
     assert stream.priority == 0
-    stream = Stream._init(options=StreamOptions(priority=-1))
+    stream = Device().create_stream(options=StreamOptions(priority=-1))
     assert stream.priority == -1
     with pytest.raises(ValueError):
-        stream = Stream._init(options=StreamOptions(priority=1))
+        stream = Device().create_stream(options=StreamOptions(priority=1))
 
-def test_stream_sync():
-    stream = Stream._init(options=StreamOptions())
+def test_stream_sync(init_cuda):
+    stream = Device().create_stream(options=StreamOptions())
     stream.sync()  # Should not raise any exceptions
 
-def test_stream_record():
-    stream = Stream._init(options=StreamOptions())
+def test_stream_record(init_cuda):
+    stream = Device().create_stream(options=StreamOptions())
     event = stream.record()
     assert isinstance(event, Event)
 
-def test_stream_record_invalid_event():
-    stream = Stream._init(options=StreamOptions())
+def test_stream_record_invalid_event(init_cuda):
+    stream = Device().create_stream(options=StreamOptions())
     with pytest.raises(TypeError):
         stream.record(event="invalid_event")
 
-def test_stream_wait_event():
-    stream = Stream._init(options=StreamOptions())
+def test_stream_wait_event(init_cuda):
+    stream = Device().create_stream(options=StreamOptions())
     event = Event._init()
     stream.record(event)
     stream.wait(event)  # Should not raise any exceptions

```

Comment:
we can actually test the real-world use case (also get rid of `Event._init()`):
```suggestion
def test_stream_wait_event(init_cuda):
    s1 = Device().create_stream()
    s2 = Device().create_stream()
    e1 = s1.record()
    s2.wait(e1)  # Should not raise any exceptions
    s2.sync()
```